### PR TITLE
Keep more specific cell styling instead of it being overwritten by the table cell_style

### PR DESCRIPTION
**Pull request by petergoldstein**
This is a rebased and updated version of #59, which addressed #56.

Thanks to @russellsanders for the original contribution.

@pointlessone I believe this can be merged.  Can you please take a look?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Master
 
+* Bugfix: Use a cell's custom style over table styles. (PR [#59](https://github.com/prawnpdf/prawn-table/pull/59), issue [#65](https://github.com/prawnpdf/prawn-table/issues/56))
```
Comment (pointlessone) [inline CHANGELOG.md:3]: Typo in issue number.

File: lib/prawn/table.rb [hunk 2]
```diff
@@ -140,7 +140,7 @@ def make_table(data, options={}, &block)
     #
     def initialize(data, document, options={}, &block)
       @pdf = document
-      @cells = make_cells(data)
+      @cells = make_cells(data, options.delete(:cell_style) || {})
```
Comment (pointlessone) [inline lib/prawn/table.rb:143]: I believe it's rude to mutate arguments.

File: CHANGELOG.md (hunk 1 above)
Comment (petergoldstein) [reply CHANGELOG.md]: @pointlessone Fixed.

**Comment by petergoldstein** [comment]
@pointlessone Review issues have been addressed, so this is likely ready for merge.

**Review by pointlessone — APPROVED** [review]

**Comment by mojavelinux** [comment]
This is a significant breaking change which affects Asciidoctor PDF. The problem is that the `:cell_style` option on the table no longer gets applied to each custom cell implementation. Before, setting `:cell_style` would be passed to `cell_style=`, which in turn would be passed to `cells.style`. That no longer happens, and `make_cells` does not pass options to a cell that's already been constructed. So, essentially, the code path has changed.

I can likely find a way to work around it (see https://github.com/asciidoctor/asciidoctor-pdf/commit/9d301228d4e83b3c0a218b3a04beeaaaa8c1dc04), but I just wanted you to be aware that the change does break widely used code when you decide how to version the release.

**Pull request merged**